core/app/models/spree/order.rb

```
@@ -510,6 +510,15 @@ def empty!
       self
     end
 
+    def advance
+      while self.next; end
+      self
+    end
+
+    def advance!
```

I agree that since there is one call side we can just do the work inline within the controller. I had experimented with using `#advance!` in a few other places, but I backed out those changes because they had unintended side effects.

As for the naming convention, I was following the guidelines in http://dablog.rubypal.com/2007/8/15/bang-methods-or-danger-will-rubyist

The basic idea is that the bang method is an "unsafe" version of some same-named non-bang method. In this case there's the safer `#advance` method which doesn't swallow errors. As I understand it, this was the original convention as laid out by Matz, but then the rails developers decided to override this and have bang mean either mutate or raises.